### DISCUSSION TITLE
Sql Server default constraints have generated names

### DISCUSSION THREAD

**Discussion by marshall62** [sqlalchemy/alembic#1414] [Usage Questions]
The use of server_default results in a generated constraint in Sql Server.   
op.add_column("OrderTable", sa.Column("item_counter", sa.Integer(), nullable=False, server_default="0"))

A constraint like df_OrderTable_item_cou__5A843B7E3. is generated when alembic upgrade is run.

At present, there is no way to gain control over the name of the constraint.  This means that the downgrade function fails
because there is no way to add a command to delete the constraint.

**Comment by zzzeek** [comment]
use [mssql_drop_default](https://alembic.sqlalchemy.org/en/latest/ops.html#alembic.operations.Operations.drop_column.params.mssql_drop_default) for this.
